pkg/util/azureclient/debug.go

```diff
@@ -0,0 +1,29 @@
+package azureclient
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+
+	"github.com/Azure/go-autorest/autorest"
+)
+
+// LoggingSender intercepts requests and logs them out.
+// Usage: client.Sender = &LoggingSender{client.Sender}
+type LoggingSender struct {
+	autorest.Sender
+}
+
+func (ls *LoggingSender) Do(req *http.Request) (*http.Response, error) {
+	b, _ := httputil.DumpRequestOut(req, true)
+	fmt.Printf("%s\n\n", string(b))
```

Comment:
I get why you're not using logrus here; `fmt.Fprintf(os.Stderr, ...` wouldn't hurt though